source/ota.c

```diff
@@ -3170,7 +3215,19 @@ OtaErr_t OTA_Init( OtaAppBuffer_t * pOtaBuffer,
         /*
          * Initialize OTA event interface.
          */
-        ( void ) otaAgent.pOtaInterface->os.event.init( NULL );
+        if( !otaAgent.isOtaInterfaceInited )
+        {
+            ( void ) otaAgent.pOtaInterface->os.event.init( NULL );
+            otaAgent.isOtaInterfaceInited = 1;
+        }
+        else
+        {
+            /* Drop all events that created after OTA_Shutdown. */
+            while( otaAgent.pOtaInterface->os.event.recv( NULL, &eventMsg, 0 ) == OtaOsSuccess )
```

Comment:
Could we have a new function  like xQueueReset() which can reset the queue atomically ?

Reply:
Function resetEventQueue added.